### ISSUE TITLE
Type conversion example does not compile

### ISSUE THREAD

**Issue by Alexander-N** [PyO3/pyo3#334]
### 🌍 Environment

 - Your operating system and version: Ubuntu 18.10
 - Your python version: 3.6.7
 - How did you install python (e.g. apt or pyenv)? apt
 - Did you use a virtualenv?: no
 - Your rust version (`rustc --version`): rustc 1.32.0-nightly (5aff30734 2018-11-19)
 - Are you using the latest pyo3 version? Have you tried using latest master (replace `version = "0.x.y"` with `git = "https://github.com/PyO3/pyo3")?` yes

### 💥 Reproducing
When trying to build the following example from https://pyo3.rs/master/conversions.html#args-and-kwargs-for-python-object-call

```rust
extern crate pyo3;

use pyo3::prelude::*;
use pyo3::types::PyDict;
use std::collections::HashMap;

struct SomeObject;
impl SomeObject {
    fn new(py: Python) -> PyObject {
        PyDict::new(py).to_object(py)
    }
}
fn main() {
    let key1 = "key1";
    let val1 = 1;
    let key2 = "key2";
    let val2 = 2;

    let gil = Python::acquire_gil();
    let py = gil.python();

    let obj = SomeObject::new(py);

    // call object with PyDict
    let kwargs = PyDict::new(py);
    kwargs.set_item(key1, val1);
    obj.call(py, NoArgs, kwargs);

    // pass arguments as rust tuple
    let kwargs = ((key1, val1), (key2, val2));
    obj.call(py, NoArgs, kwargs);

    // pass arguments as HashMap
    let mut kwargs = HashMap::<&str, i32>::new();
    kwargs.insert(key1, 1);
    obj.call(py, NoArgs, kwargs);
}
```
I get 
```
error[E0308]: mismatched types
  --> src/main.rs:27:26
   |
27 |     obj.call(py, NoArgs, kwargs);
   |                          ^^^^^^
   |                          |
   |                          expected enum `std::option::Option`, found reference
   |                          help: try using a variant of the expected type: `Some(kwargs)`
   |
   = note: expected type `std::option::Option<&pyo3::types::PyDict>`
              found type `&pyo3::types::PyDict`

error[E0308]: mismatched types
  --> src/main.rs:31:26
   |
31 |     obj.call(py, NoArgs, kwargs);
   |                          ^^^^^^ expected enum `std::option::Option`, found tuple
   |
   = note: expected type `std::option::Option<&pyo3::types::PyDict>`
              found type `((&str, {integer}), (&str, {integer}))`

error[E0308]: mismatched types
  --> src/main.rs:36:26
   |
36 |     obj.call(py, NoArgs, kwargs);
   |                          ^^^^^^ expected enum `std::option::Option`, found struct `std::collections::HashMap`
   |
   = note: expected type `std::option::Option<&pyo3::types::PyDict>`
              found type `std::collections::HashMap<&str, i32>`

error: aborting due to 3 previous errors
```

Passing Options and replacing the tuple with a Vec makes it compile.
```rust
extern crate pyo3;

use pyo3::prelude::*;
use pyo3::types::{IntoPyDict, PyDict};
use std::collections::HashMap;

struct SomeObject;
impl SomeObject {
    fn new(py: Python) -> PyObject {
        PyDict::new(py).to_object(py)
    }
}
fn main() {
    let key1 = "key1";
    let val1 = 1;
    let key2 = "key2";
    let val2 = 2;

    let gil = Python::acquire_gil();
    let py = gil.python();

    let obj = SomeObject::new(py);

    // call object with PyDict
    let kwargs = PyDict::new(py);
    kwargs.set_item(key1, val1);
    obj.call(py, NoArgs, Some(kwargs));

    // pass arguments as Vec
    let kwargs = vec![(key1, val1), (key2, val2)];
    obj.call(py, NoArgs, Some(kwargs.into_py_dict(py)));

    // pass arguments as HashMap
    let mut kwargs = HashMap::<&str, i32>::new();
    kwargs.insert(key1, 1);
    obj.call(py, NoArgs, Some(kwargs.into_py_dict(py)));
}
```

**Comment by konstin** [comment]
Thank you for the report and the fix, I've included that in #336.

In general we'd like to have the examples tested automatically, but that is blocked on https://github.com/assert-rs/docmatic/issues/4.

**Comment by Alexander-N** [comment]
Is it intentional that `IntoPyDict` is implemented for `Vec` but not for tuples?

Wouldn't it be possible to use the following workaround from assert-rs/docmatic#4 until the bug is fixed?
```bash
rm -rf target/debug/deps/*pyo3* 
```
